server/assetstore.py

```diff
@@ -562,16 +630,58 @@ def validateFile(file):
     """
     if DB_INFO_KEY not in file or 'assetstoreId' not in file:
         return None
-    assetstore = ModelImporter.model('assetstore').load(
-        file['assetstoreId'])
+    assetstore = Assetstore().load(file['assetstoreId'])
     if assetstore.get('type') != AssetstoreType.DATABASE:
+        # This can happen if the file was a database_assetstore file and then
+        # is replaced, for instance, by uploading a new file.
+        if DB_INFO_KEY in file:
+            del file[DB_INFO_KEY]
         return None
     if not file[DB_INFO_KEY].get('table'):
         raise ValidationException(
             'File database information entry must have a non-blank table '
             'value.')
-    if (not dbs.databaseFromUri(assetstore['database']['uri']) and
-            not file[DB_INFO_KEY].get('database')):
+    if not assetstore['database'].get('uri') and not file[DB_INFO_KEY].get('uri'):
         raise ValidationException(
-            'File database information must have a non-blank database value '
-            'on an assetstore that doesn\'t specify a single database.')
+            'File database information must have a non-blank uri value on an '
+            'assetstore that doesn\'t specify a single database.')
+
+
+def checkUserImport(user, uri, validateUri=True):
+    """
+    Check if the specified user is allowed to import the specified URI.
+
+    :param user: the user to validate.
+    :param uri: the database connection string to validate.
+    :param validateUri: if False, only report if the user could import any
+        database (the uri parameter is always considered matching).
+    :returns: None if the user may import the specified URI, otherwise an error
+        string.
+    """
+    if not Setting().get(PluginSettings.USER_DATABASES):
+        return 'User-level database imports are not allowed'
+    groups = Setting().get(PluginSettings.USER_DATABASES_GROUPS)
+    # Admins can always do this if the base setting allows it
+    if user.get('admin'):
+        return
+    if groups and len(groups):
```

Comment:
Can just be `if groups:` since that would only be truthy if `len(groups)` was also truthy.